umbral/umbral.py

```diff
@@ -421,3 +436,55 @@ def decapsulate_reencrypted(self, pub_key: Point, priv_key: BigNum, orig_pub_key
         assert (s * inv_d) * orig_pub_key == (h * e_prime) + v_prime, "Generic Umbral Error"
 
         return key
+
+    def encrypt(self, pub_key: UmbralPublicKey, data: bytes):
+        """
+        Performs an encryption using the UmbralDEM object and encapsulates a key
+        for the sender using the public key provided.
+
+        Returns the ciphertext and the KEM Capsule.
+        """
+        key, capsule = self._encapsulate(pub_key.point_key, SecretBox.KEY_SIZE)
+
+        dem = UmbralDEM(key)
+        enc_data = dem.encrypt(data)
+
+        return enc_data, capsule
+
+    def decrypt(self, priv_key: UmbralPrivateKey, capsule: Capsule, enc_data: bytes):
+        """
+        Decrypts the data provided by decapsulating the provided capsule.
+
+        Returns the plaintext of the data.
+        """
+        key = self._decapsulate_original(
+            priv_key.bn_key, capsule
+        )
+
+        dem = UmbralDEM(key)
+        plaintext = dem.decrypt(enc_data)
+
+        return plaintext
+
+    def decrypt_reencrypted(self, recp_priv_key: UmbralPrivateKey,
+                            recp_pub_key: UmbralPublicKey,
+                            sender_pub_key: UmbralPublicKey,
+                            capsule: Capsule, enc_data: bytes):
+        """
+        Decrypts the data provided by reconstructing and decapsulating the
+        capsule.
+
+        Returns the plaintext of the data.
+        """
+        # TODO: Do we take only the recipient's private key and form pubkey?
```

Comment:
Yes, we can do that, since the public key can be deterministically derived from the private key.